src/ucp/wireup/address.c

```diff
@@ -390,11 +396,19 @@ static int ucp_address_pack_iface_attr(ucp_worker_h worker, void *ptr,
     packed->bandwidth      = iface_attr->bandwidth.dedicated - iface_attr->bandwidth.shared;
     packed->lat_ovh        = iface_attr->latency.overhead;
 
-    /* Keep only the bits defined by UCP_ADDRESS_IFACE_FLAGS to shrink address. */
+    /* Keep only the bits defined by UCP_ADDRESS_IFACE_FLAGS
+     * to shrink address. */
     packed->prio_cap_flags |=
         ucp_address_pack_flags(iface_attr->cap.flags,
                                UCP_ADDRESS_IFACE_FLAGS, 8);
 
+    /* Keep only the bits defined by UCP_ADDRESS_IFACE_EVENT_FLAGS
+     * to shrink address. */
+    packed->prio_cap_flags |=
+        ucp_address_pack_flags(iface_attr->cap.event_flags,
+                               UCP_ADDRESS_IFACE_EVENT_FLAGS,
+                               8 + ucs_popcount(UCP_ADDRESS_IFACE_FLAGS));
```

Comment:
can we check it does not exceed size (22) ?

Reply:
done